extension/bearertokenauthextension/bearertokenauth.go

```diff
@@ -44,26 +47,99 @@ func (c *PerRPCAuth) RequireTransportSecurity() bool {
 
 // BearerTokenAuth is an implementation of configauth.GRPCClientAuthenticator. It embeds a static authorization "bearer" token in every rpc call.
 type BearerTokenAuth struct {
-	tokenString string
-	logger      *zap.Logger
+	muTokenString sync.RWMutex
+	tokenString   string
+
+	shutdownCH chan struct{}
+
+	tokenFilename string
+	logger        *zap.Logger
 }
 
 var _ configauth.ClientAuthenticator = (*BearerTokenAuth)(nil)
 
 func newBearerTokenAuth(cfg *Config, logger *zap.Logger) *BearerTokenAuth {
+	if cfg.BearerTokenFilename != "" && cfg.BearerToken != "" {
+		logger.Warn("a filename is specified. Configured token is ignored!")
+	}
 	return &BearerTokenAuth{
-		tokenString: cfg.BearerToken,
-		logger:      logger,
+		tokenString:   cfg.BearerToken,
+		tokenFilename: cfg.BearerTokenFilename,
+		logger:        logger,
 	}
 }
 
-// Start of BearerTokenAuth does nothing and returns nil
+// Start of BearerTokenAuth does nothing and returns nil if no tokenFilename
+// is specified. Otherwise a routine is started to monitor the file containing
+// the token to be transferred.
 func (b *BearerTokenAuth) Start(ctx context.Context, host component.Host) error {
-	return nil
+	if b.tokenFilename == "" {
+		return nil
+	}
+
+	if b.shutdownCH != nil {
+		return fmt.Errorf("bearerToken file monitoring is already running")
```

Comment:
When would this be the case?

Reply:
The shutdown channel is created when the extension is started and removed when stop is called. This channel condition is used to prevent another watcher from being spawned.

Reply:
Is this happening when the component is started twice without being shut down?

Reply:
Hopefully never. But yes, in case it get started twice, before it was stopped.